lib/models/os.coffee

```diff
@@ -85,21 +85,12 @@ getOsModel = (deps, opts) ->
 		else
 			version
 
-	unfixNonSemver = (version) ->
-		if version?
-			version.replace(/\+FIXED-rev(\d+)/, '.rev$1')
-		else
-			version
-
 	# utility method exported for testability
 	exports._getMaxSatisfyingVersion = (versionOrRange, osVersions) ->
 		if versionOrRange in [ 'default', 'latest', 'recommended' ]
 			return osVersions[versionOrRange]
 
-		semverVersions = osVersions.versions.map(fixNonSemver)
-		maxVersion = semver.maxSatisfying(semverVersions, fixNonSemver(versionOrRange))
-
-		return unfixNonSemver(maxVersion)
+		return semver.maxSatisfying(semverVersions, fixNonSemver(versionOrRange))
```

Comment:
Can we move `fixNonSemver` handling of the range into `resin-semver` too? It'd be nice to be able to remove that from here entirely.

Reply:
I think this is functionally equivalent to https://github.com/resin-io-modules/resin-semver/blob/master/src/index.js#L11 what do you think?

Reply:
There are potential differences, but I do agree it's close enough that it should work for everything we care about - i.e. if you just safeSemver the range inside resin-semver, I think everything'll be fine.

Note that @thgreasi is in the process of drastically updating that function (https://github.com/resin-io-modules/resin-semver/pull/35/files#diff-f41e9d04a45c83f3b6f6e630f10117fe). I think this is still fine though :+1:.